enhancements/managedClustersUpgrade.md

```diff
@@ -0,0 +1,592 @@
+---
+title: Managed Clusters status enhancement
+
+authors:
+  - @serngawy
+
+reviewers:
+  - TBD
+
+approvers:
+  - @ijolliffe
+
+api-approvers:
+  - TBD
+
+creation-date: 2022-06-28
+
+last-updated: 2022-06-28
+
+see-also:
+   - TBD
+
+tracking-link:
+   - TBD
+---
+
+
+# Cluster Group Upgrade (CGU) and Managed Cluster upgrade (MCU) status enhancement
+
+
+## Summary
+
+Cluster Group upgrade (CGU) and Managed Cluster Upgrade (MCU) provide aggregator  APIs for managed cluster upgrades and configuration change. CGU uses ACM policy APIs to apply cluster configuration changes and present the status of the policies per managed cluster. MCU uses ACM manifestwork APIs to upgrade ocp clusters and present the status of the upgrade process per managed cluster.
+
+## Motivation
+
+1. Currently The CGU status does not provide state per managed cluster. In case of failure to apply the ACM policies, the end user cannot identify which managed cluster failed. The enhancement proposes a change to the CGU status to include policy state per managed cluster.
+
+1. CGU uses ACM policy to upgrade an ocp cluster and operators. The ACM policy provides a binary state (Compliant/NonCompliant) while the upgrade process takes at least 40min to 2h and it has 5 different states; NotStarted, Initialized, InProgress (with gradual percentage), Complete or Failed. MCU will use ACM manifest work API in order to provide better state report for the upgrade process while the CGU will continue to use ACM policy API to apply cluster configuration changes.
+
+### User Stories
+
+1. As an end user, I would like to enforce a set of ACM policies to a group of clusters and be able to track the policies state compliant/NonCompliant per cluster.
+
+1. As an end user, I would like to upgrade a group of clusters and be able to track the upgrade process state per cluster.
+
+
+### Goals
+
+1. Enhance the CGU status to present the state of ACM policies per managed clusters
+
+1. Provide a managed clusters upgrade APIs to present the upgrade process state per managed cluster
```

Comment:
its not possible to include that in the same CGU status. Plus as i mentioned using policy does not give the ability to to have a progress state.

Reply:
Why does policy not allow for a progress state? The TALM controller, when it sees that it is managing a ClusterVersion Policy (it already detects subscriptions and has special handling), can run specific logic to pull the upgrade status from the ClusterVersion CR on a periodic basis.

Reply:
Having two different controllers / APIs is confusing to me, and might confuse our users. It could have been better if it was an isolated change, but it also moves stuff from CGU to MCU, giving our users grief. Can we try and design the CGU API of our dreams first, and then discuss which underlying technology we use to fulfill it? 
For example, we can extend the existing API in a non-disruptive way with `detailed-status`:

```yaml
- name: spoke3
   policies:
     policy1-common: compliant
     policy2-group: in progress
     policy3-site:  notApplied
   state: failed
   detailed-status:
     backup: 
       message: succeeded in 0 minutes 12 seconds
     pre-caching:
       message:  succeeded in 40 minutes 33 seconds
       per-cluster-spec: {if we ever decide to implement it}
       images-pre-cached: 502 of 502
       size-on-disk: 12.6 GiB
       free-space-percent: 32
     policies:
     - policy1-common:
       message: compliant 
     - policy2-group:
       message: waiting object performance.openshift.io/v2/performanceprofiles to comply, node NotReady
```
We can implement it using one underlying technology today and switch to another one tomorrow if we find it worthy

Reply:
I agree with @vitus133, it looks more cleaner to be in the same place i.e. at CGU

Reply:
my concern about having all spec together in the same CR is the status will be different based on the process and it might be very long.